python_examples/age_stratified_example/age_stratified_plot.py

```diff
@@ -0,0 +1,176 @@
+                 age_list: list = None):
+        """Initialise the plotter with a filepath and read
+        data from the csv.
+        Parameters
+        ----------
+        filepath : str
+            Filepath to the .csv containing output data
+        start_date : str
+            Starting date for the simulation, "day-month-year"
+        sum_weekly : bool
+            Flag to plot either each timepoint, or the 7 day sum
+        age_list : list
+            List of the explicit age ranges saved in the csv
+        """
+        if not os.path.splitext(filepath)[1] == ".csv":
+            raise TypeError("input file" + filepath + "must be .csv")
+        self.data = pd.read_csv(filepath)
+        self.age_list = age_list
+        self.start_date = start_date
+        self.sum_weekly = sum_weekly
+
+        # Identify the column containing age stratifcation as any
+        # with the substring "age"
+        self.age_name = next((s for s in list(self.data.columns)
+                             if "age" in s.lower()), None)
+        self.do_ages = (self.age_name is not None)
+        if self.do_ages and (self.age_list is None):
+            num = self.data[self.age_name].max()
+            self.age_list = [str(5*i)+"-"+str(5*i+5) for i in range(num+1)]
+
+    def sum_infectious(self) -> None:
+        """Helper function which sums across all columns containing infectious
+        people and appends the dataframe with a further column containing this
+        data.
+        """
+        total = self.data[list(self.data.
+                               filter(regex='InfectionStatus.Infect'))]
+        self.data["Total Infectious"] = total.sum(axis=1)
+
+    def dates(self, dataFrame=None, period='weekly') -> None:
+        """Helper function to calculate a dictionary associating each
+        timestep, with the date of a day, or week.
+        Parameters
+        ----------
+        dataFrame : DataFrame
+            dataFrame containing a "time" column
+        period : str
+            Either "daily" or "weekly"
+        Returns
+        -------
+        date_list: list
+            List of dates, corresponding to the time column
+        """
+        assert self.start_date is not None, 'Start date not set'
+        assert period in ['daily', 'weekly'], 'Period not daily or weekly'
+        if dataFrame is None:
+            dataFrame = self.data
+        date_list = []
+        timepoints = list(dataFrame['time'])
+        start = pd.to_datetime(self.start_date).date()
+        for time in timepoints:
+            if period == 'daily':
+                date_list.append((start + pd.DateOffset(days=time)).date())
+            else:
+                if time % 7 == 0:
+                    week_date = (start + pd.DateOffset(days=time)).date()
+                date_list.append(week_date)
+        date_list = [d.strftime('%m-%d') for d in date_list]
+        return date_list
+
+    def fiveToTen(self):
```

Comment:
Function name does not obviously indicate what function does (i.e. Convert between/combine different age groups)

Reply:
:(